apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx

```diff
@@ -33,16 +33,37 @@ export default function CodeReviewGroupsManager({initialGroups}) {
 
   const onGroupDelete = droppableId => {
     // First, take all group members from deleted group and put them in unassigned group
-    const updatedUnassignedGroup = unassignAllFromGroup(droppableId);
+    const updatedUnassignedGroup = unassignAllFromGroup(
+      getUnassignedGroup(),
+      getGroup(droppableId)
+    );
     const updatedGroups = updateGroups(groups, [updatedUnassignedGroup]);
 
     setGroups(updatedGroups.filter(group => group.droppableId !== droppableId));
   };
 
-  const unassignAllFromGroup = droppableId => {
-    const updatedUnassignedGroup = {...getUnassignedGroup()};
-    const unassignedGroup = getGroup(droppableId);
-    updatedUnassignedGroup.members.push(...unassignedGroup.members);
+  const unassignAll = () => {
+    let updatedUnassignedGroup = _.cloneDeep(getUnassignedGroup());
+    getAssignedGroups().forEach(group => {
+      updatedUnassignedGroup = unassignAllFromGroup(
+        updatedUnassignedGroup,
+        group
+      );
+    });
+
+    const updatedAssignedGroups = getAssignedGroups().map(group => ({
+      ...group,
+      members: []
+    }));
+    setGroups([...updatedAssignedGroups, updatedUnassignedGroup]);
+  };
+
+  const unassignAllFromGroup = (
```

Comment:
nit: could this be something like `copyToUnassigned` maybe, just to be clear that this is only moving students to the unassigned group, not deleting them from another group?

Or alternatively, what if this function called `updateGroups` with the updated unassigned group and updated empty group and just returned the full list of groups? That way you'd be unassigning all from a single group in a single step?

Reply:
Yeah I felt like the naming wasn't great here / logic a bit confusing, I think these are good ideas. Let me give it a shot.

Reply:
I renamed per your first suggestion -- the second idea makes sense to me, but I've been a little inconsistent in where I copy groups before changing them and would need to make a few changes to make that work. I am planning on pulling a lot of these get/change/etc methods out into a little util file, so I'm thinking I could maybe clean this up at that point? Makes sense to me though, as I'm looping through assigned groups twice